### PR TITLE
Secbots no longer attack people from inside locked containers

### PR DESCRIPTION
**Pull request by Superlagg**
[BUGFIX] [FEATURE]

## About the PR 

If a secbot is welded into a locker (or any /obj/storage), they'll thrash around inside it for a while, then give up, dropping their target and returning to being idle.

While they will thrash around and make a lot of noise, they aren't able to actually break free of a locked/welded locker. They can escape unlocked lockers just fine, though.

They can still see from inside lockers, just like us. They'll react to perps, get angry, and, if they're freed, engage that perp.

## Why's this needed? 

Secbots keep stunning people from inside of things they shouldn't. Now they can't! 

## Changelog

```changelog
(u)Superlagg
(+)Securitrons locked or welded into a locker can no longer attack people. They aren't happy about this, though.
```

### CODE REVIEW

**Review by Tarmunora — APPROVED** [review]

File: code/modules/robotics/bot/secbot.dm [hunk 1]
```diff
@@ -741,6 +745,34 @@
...
+							M.show_text("<font size=[max(0, 5 - get_dist(get_turf(src), M))]>THUD, thud!</font>")
+						playsound(get_turf(C), "sound/impact_sounds/Wood_Hit_1.ogg", 15, 1, -3)
+						var/wiggle = 6
+						while(wiggle > 0)
+							wiggle--
+							C.pixel_x = rand(-3,3)
+							C.pixel_y = rand(-3,3)
+							sleep(0.1 SECONDS)
+						C.pixel_x = initial(C.pixel_x)
+						C.pixel_y = initial(C.pixel_y)
```
Comment (pali6) [inline code/modules/robotics/bot/secbot.dm:765]:
```suggestion
						animate_storage_thump(C)
```

File: code/modules/robotics/bot/secbot.dm [hunk 2]
```diff
@@ -843,26 +875,27 @@
 		weeoo()
 		process()	// ensure bot quickly responds to a perp
 
-	proc/YellAtPerp()
-		src.point(src.target, 1)
-		src.speak("Level [src.threatlevel] infraction alert!")
+	proc/YellAtPerp(impotent_rage = 0)
```
Comment (pali6) [inline code/modules/robotics/bot/secbot.dm:878]: How does this relate to the rest of the PR? I don't see any calls of `YellAtPerp` being added nor modified?

**Review by pali6 — CHANGES_REQUESTED** [review]

File: code/modules/robotics/bot/secbot.dm (hunk 2 above)
Comment (Superlagg) [reply code/modules/robotics/bot/secbot.dm:871]: Oh yeah, the bot was supposed to occasionally say things while locked inside. I took it out cus it felt kinda dumb and forgot to clean it up. I'll clean it up!

Comment (Superlagg) [reply code/modules/robotics/bot/secbot.dm]: Okay I think I got all the impotent rage out.

**Comment by Superlagg** [comment]
I also moved the cuff bar failure checks to its own proc, and added in a check to also fail if either the bot or the target are in a container, so they won't cuff people (from) inside lockers or pods.

**Review by Tarmunora — APPROVED** [review]

**Review by Sovexe — CHANGES_REQUESTED** [review]

**Pull request merged**